day-one-challenge/tyranny-of-the-rocket/getFuelRequirements.js

```diff
@@ -0,0 +1,20 @@
+const { sum, map } = require('ramda');
+const moduleMassArray = require('./fuelData');
+
+const calculateFuel = mass => {
+  return Math.floor(mass / 3) - 2;
+};
+
+const fuelOfCurrentModuleOfMass = (currentMass) => {
+  if (typeof (currentMass) !== 'number')
+    throw new TypeError("mass entered is not a number!");
+  return calculateFuel(currentMass) <= 0 ? 0 : calculateFuel(currentMass) + fuelOfCurrentModuleOfMass(calculateFuel(currentMass));
+}
+
+const totalFuelForAllModulesOfMass = array => {
+  return sum(map(fuelOfCurrentModuleOfMass, array));
+};
+
+console.log(totalFuelForAllModulesOfMass(moduleMassArray));
```

Comment:
When this module is required (in your tests, for example) this line will always run. What about creating a driver for this code (maybe `index.js`)? Then your module would just be focused on calculating masses.

Reply:
Thanks, this is a great idea! I have implemented it here: 6375a93eb586c9951ea68ca30fe20508d14c22d8.